src/generator/AutoRest.Go/Extensions.cs

```diff
@@ -89,6 +91,26 @@ public static string[] ToWords(this string value)
             return SplitPattern.Split(value).Where(s => !string.IsNullOrEmpty(s)).ToArray();
         }
 
+        public static string Singularize(this string value)
+        {
+            PluralizationService ps = PluralizationService.CreateService(CultureInfo.GetCultureInfo("en-us"));
+            return ps.Singularize(value);
+        }
+
+        /// <summary>
+        /// This method checks if MethodGroupName is plural of package name.
+        /// if yes, it does not remove package name from group name.
```

Comment:
looks like this method only checks so this comments seems appropriate on the above method.